nxpprog.py

```diff
@@ -398,6 +399,80 @@
 prog = None
 
 
+LPC40XX_CHAR = {'Question': b'?',
+                'Synchronized': b'Synchronized\r\n',
+                'OK': b'OK\r\n'}
+
+
+def port_write_and_verify(port, payload, error_message="", debug_message=""):
+    bytes_sent = port.write(bytearray(payload))
+    if bytes_sent != len(payload):
+        log(error_message)
+        return False
+    else:
+        log(debug_message)
+        return True
+
+
+def port_read(port, number_of_bytes):
+    return bytearray(port.read(number_of_bytes))
+
+
+def attempt_sync(port):
+    log(f'Attempting to sync with device at {port.port}')
+    time.sleep(.100)
+    port.dtr = True
+    port.rts = True
+    time.sleep(.100)
+    port.dtr = False
+    time.sleep(.100)
+    port.rts = False
+    time.sleep(1)
+    port_write_and_verify(port, LPC40XX_CHAR['Question'])
+    time.sleep(.100)
+    response = port_read(port, 14)
+    time.sleep(.100)
+    port_write_and_verify(port, LPC40XX_CHAR['Synchronized'])
+    time.sleep(.100)
+    response = port_read(port, 17)
+    port_write_and_verify(port, b'10000\r\n')
```

Comment:
What is 10000? Is that a baud rate? If it is, you should use something more typical like 115200.

Reply:
Done.